### PR TITLE
Error handling cleanup in solvers/sat

### PR DESCRIPTION
**Pull request by danpoe**

### CODE REVIEW

File: src/solvers/sat/resolution_proof.cpp [hunk 1]
```diff
@@ -39,19 +40,20 @@ void resolution_prooft<T>::build_core(std::vector<bool> &in_core)
...
-        assert(v<in_core.size());
+        INVARIANT(
+          v < in_core.size(), "variable number should be within bounds");
         in_core[v]=true;
       }
     }
     else
     {
-      assert(c.first_clause_id<c_id);
+      INVARIANT(c.first_clause_id < c_id, "");
```
Comment (peterschrammel) [inline src/solvers/sat/resolution_proof.cpp]: missing message

File: src/solvers/sat/resolution_proof.cpp [hunk 2]
```diff
@@ -39,19 +40,20 @@ void resolution_prooft<T>::build_core(std::vector<bool> &in_core)
...
       s.push(c.first_clause_id);
 
       for(clauset::stepst::size_type i=0; i<c.steps.size(); i++)
       {
         // must decrease
-        assert(c.steps[i].clause_id<c_id);
+        INVARIANT(c.steps[i].clause_id < c_id, "");
```
Comment (peterschrammel) [inline src/solvers/sat/resolution_proof.cpp]: missing message

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: df857ffc7e6269d8e187bb58e5385a95451368b7).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85589093

**Comment by danpoe** [comment]
@peterschrammel @tautschnig I think the PR is ready now.

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**